### PR TITLE
Bump to Android Studio Flamingo

### PR DESCRIPTION
**Pull request by wmontwe**
Change project setup to use Android Studio Flamingo with Android Gradle Plugin 8.0. This also bumps Kotlin to 1.8.20 and Gradle to 8.1.1

See [Android Studio Flamingo is stable](https://android-developers.googleblog.com/2023/04/android-studio-flamingo-is-stable.html)

Fixes: #6703

### CODE REVIEW

File: .java-version
[the remark this answers is not in the document]
Comment (wmontwe) [reply .java-version:1]: That's [jenv: Manage your Java environment](https://github.com/jenv/jenv)

File: build-plugin/src/main/kotlin/ThunderbirdProjectConfig.kt [hunk 1]
```diff
@@ -2,7 +2,8 @@ import org.gradle.api.JavaVersion
 
 object ThunderbirdProjectConfig {
 
-    val javaVersion = JavaVersion.VERSION_11
+    val javaVersion = JavaVersion.VERSION_17
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply build-plugin/src/main/kotlin/ThunderbirdProjectConfig.kt:5]: This is required by the compose compiler 1.4.5 version. There will be a new release [1.4.6](https://developer.android.com/jetpack/androidx/releases/compose-compiler#1.4.5) that should fix it.

File: build.gradle.kts [hunk 2]
```diff
@@ -72,10 +71,6 @@ allprojects {
         }
     }
 
-    tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
```
Comment (cketti) [inline build.gradle.kts:75]: With this removed, can we also get rid of the following lines further up in the file?

```kotlin
val javaVersion = JavaVersion.VERSION_11
val jvmTargetVersion = JvmTarget.JVM_11
```

**Review by cketti — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (wmontwe) [reply build.gradle.kts:75]: Oh, it's in a later commit. Will remove it here.

**Pull request merged**